Desktop/src/renderer/AddNodesDnD/AddButton.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './index.css'
+import { CSSTransition } from 'react-transition-group'
+import { BsPlusLg, BsPlayFill, BsChevronRight, BsArrowLeft, BsArrowBarUp, BsArrowBarDown, BsFillHddRackFill, BsFillDiagram2Fill } from "react-icons/bs";
+import Subscribe from './NodeInput/Subscribe';
+import Application from './NodeInput/Application';
+import Publish from './NodeInput/Publish';
+
+
+const AddButton = ({ nodes, setNodes }) => {
+
+    return (
+        <Navbar>
+
+            <NavItem icon={<BsPlayFill />} />
+
+            <NavItem icon={<BsPlusLg />}>
+                <DropdownMenu nodes={nodes} setNodes={setNodes} ></DropdownMenu>
+            </NavItem>
+
+        </Navbar>
+    );
+}
+
+function Navbar(props: { children: React.ReactNode }) {
+    return (
+        <nav className="navbar">
+            <ul className="navbar-nav">{props.children}</ul>
+        </nav>
+    );
+}
+
+function NavItem(props: { icon: React.ReactChild; children?: React.ReactNode }) {
+    const [open, setOpen] = useState(false);
+
+    return (
+        <li className="nav-item">
+            <a href="#" className="icon-button" onClick={() => setOpen(!open)}>
+                {props.icon}
+            </a>
+
+            {open && props.children}
+        </li>
+    );
+}
+
+function DropdownMenu({ nodes, setNodes }) {
+    const [activeMenu, setActiveMenu] = useState('main');
+    const [menuHeight, setMenuHeight] = useState(null);
+    const dropdownRef = useRef(null);
+
+    useEffect(() => {
+        setMenuHeight(dropdownRef.current?.firstChild.offsetHeight)
+    }, [])
+
+    function calcHeight(el) {
+        const height = el.offsetHeight;
+        setMenuHeight(height);
+    }
+
+    function DropdownItem(props) {
+        return (
+            <a href="#" className="menu-item" onClick={() => props.goToMenu && setActiveMenu(props.goToMenu)}>
```

Comment:
When the menu item components are dragged in the canvas the application throws an unhandled exception. You should make the menu-item class items not be draggable at all.

Reply:
try dragging into the canvas the add nodes option in the menu it will crash the application